src/main/java/sirius/biz/process/Processes.java

```diff
@@ -482,10 +486,25 @@ protected boolean setStateMessage(String processId, String state) {
      */
     public boolean addLink(String processId, ProcessLink link) {
         return modify(processId,
-                      process -> process.getState() != ProcessState.TERMINATED,
+                      process -> true,
                       process -> process.getLinks().add(link));
     }
 
+    /**
+     * Adds the given reference to the given process.
+     *
+     * @param processId the process to update
+     * @param reference the reference to attach
+     * @return <tt>true</tt> if the process was successfully modified, <tt>false</tt> otherwise
+     */
+    public boolean addReference(String processId, String reference) {
+        return modify(processId, process -> true, process -> {
+            if (!process.getReferences().contains(reference)) {
```

Comment:
shouldn't we do this check in the "checker" lambda? (currently `process -> true`)

Reply:
The checker is in charge of ensuring the correct state, not for enforcing business rules...